pypeit/spectrographs/mdm_modspec.py

```diff
@@ -0,0 +1,317 @@
+            fitstbl (`astropy.table.Table`_):
+                The table with the metadata for one or more frames to check.
+                This table uses the Pypeit-specific metadata keywords, as defined
+                under def init_meta(self).
+            exprng (:obj:`list`, optional):
+                Range in the allowed exposure time for a frame of type
+                ``ftype``. See
+                :func:`pypeit.core.framematch.check_frame_exptime`.
+
+        Returns:
+            `numpy.ndarray`_: Boolean array with the flags selecting the
+            exposures in ``fitstbl`` that are ``ftype`` type frames.
+        """
+        good_exp = framematch.check_frame_exptime(fitstbl['exptime'], exprng)
+        if ftype in ['science']:                # Standards and Sciences lumped together under 'science'
+            return good_exp & (fitstbl['idname'] == 'Object') & (fitstbl['mirror'] == 'OUT')
+        
+        if ftype == 'bias':
+            return good_exp & (fitstbl['idname'] == 'Bias')
+        
+        if ftype in ['arc','tilt']:             # Lamps and Arcs
+            return good_exp & np.array([target in ['Comp','Object'] for target in fitstbl['idname']]) & (fitstbl['mirror'] == 'IN')
+        
+        if ftype in ['pixelflat']:              # Internal Flats
+            return good_exp & (fitstbl['idname'] == 'Flat') & (fitstbl['mirror'] == 'IN')
+                
+        if ftype in ['illumflat', 'trace']:     # Twilight Flats
+            return good_exp & (fitstbl['idname'] == 'Flat') & (fitstbl['mirror'] == 'OUT')
+        
+        msgs.warn('Cannot determine if frames are of type {0}.'.format(ftype))
+
+        return np.zeros(len(fitstbl), dtype=bool)
+    
+    def bpm(self, filename, det, shape=None, msbias=None):
+        """
+        Generate a default bad-pixel mask.
+
+        Even though they are both optional, either the precise shape for
+        the image (``shape``) or an example file that can be read to get
+        the shape (``filename`` using :func:`get_image_shape`) *must* be
+        provided.
+
+        Args:
+            filename (:obj:`str` or None):
+                An example file to use to get the image shape.
+            det (:obj:`int`):
+                1-indexed detector number to use when getting the image
+                shape from the example file.
+            shape (tuple, optional):
+                Processed image shape
+                Required if filename is None
+                Ignored if filename is not None
+            msbias (`numpy.ndarray`_, optional):
+                Master bias frame used to identify bad pixels
+
+        Returns:
+            `numpy.ndarray`_: An integer array with a masked value set
+            to 1 and an unmasked value set to 0.  All values are set to
+            0.
+        """
+        # Call the base-class method to generate the empty bpm
+        bpm_img = super().bpm(filename, det, shape=shape, msbias=msbias)
```

Comment:
Is this necessary? Could we just delete this function and it will automatically inherit the parent's routine?

Reply:
good call!